src/libexpr/eval.cc

```diff
@@ -781,7 +781,7 @@ void printStaticEnvBindings(const SymbolTable & st, const StaticEnv & se)
 // just for the current level of Env, not the whole chain.
 void printWithBindings(const SymbolTable & st, const Env & env)
 {
-    if (env.type == Env::HasWithAttrs) {
+    if (!env.values[0]->isThunk()) {
```

Comment:
Pre-existing condition, which can be considered out of scope:
This function is for the debugger. I think we'll want to print at least something if this is still a thunk. Otherwise the user may think the debugger is broken because their variable is missing.